allocate.go

```diff
@@ -30,6 +31,8 @@ type Allocator interface {
 	// Cancelling the allocator context will already perform this operation,
 	// so normally there's no need to call Wait directly.
 	Wait()
+
+	getDialHeader() http.Header
```

Comment:
Sure we need this private method in the Interface? This is usually for external use, internally we should have access to the real implementation ad not through the interface